choral/libs/choral-reactive/src/main/java/choral/reactive/ReactiveServer.java

```diff
@@ -58,19 +77,21 @@ public void listen(String address) throws URISyntaxException, IOException {
 
     @SuppressWarnings("unchecked")
     @Override
-    public <T extends Serializable> T recv(Session session) {
+    public <T extends Serializable> T recv(Session session) { // TODO profile this
         Attributes attributes = Attributes.builder().put("channel.service", serviceName)
                 .put("channel.sender", session.senderName()).build();
 
         TelemetrySession telemetrySession;
         CompletableFuture<Serializable> future = new CompletableFuture<Serializable>()
                 .orTimeout(10, TimeUnit.SECONDS);
+                // TODO Do we want this timeout? What happens if the sender takes >10s to compute the message?
+                //  Maybe the timeout is a parameter we pass *for each session*?
```

Comment:
We could easily do that, but do we need it for the paper?

Reply:
You're right, we don't need it for the paper :)